websites/M/Musixmatch/presence.ts

```diff
@@ -1,45 +1,100 @@
+			//if user status = curator show curator badge
+			if (
+				document.querySelector(
+					"#site > div > div > div > main > .mxm-page > .mxm-user-header > .container > .row > div > div > div > span"
+				)
+			) {
+				presenceData.smallImageKey = "https://i.imgur.com/Bl7BSIW.png";
+				presenceData.smallImageText = "Curator";
+			}
+			if (pathname === "/profile/me")
+				presenceData.details = "Viewing their profile";
+			else {
+				presenceData.details = `Viewing ${
+					document.querySelector("#site h1").textContent
+				}'s profile`;
+			}
+		} else if (pathname.includes("/search")) {
+			presenceData.details = "Searching";
+			presenceData.state = pathname.split("/")[2].replaceAll("%20", " ");
+		} else if (
+			pathname.includes("/lyrics") &&
+			pathname.includes("/translation/")
+		) {
+			presenceData.largeImageKey = document.querySelector<HTMLMetaElement>(
+				'[property="og:image"]'
+			).content;
+			presenceData.smallImageKey = "https://i.imgur.com/Yb4CWnm.png";
+			presenceData.details = `Reading ${pathname.split("/")[5]} translation`;
+			presenceData.state = `${
+				document.querySelector("#site h2 > span > a")?.textContent ||
+				document.querySelector("#site h2 > span > span > a").textContent
+			} - ${document.querySelector("#site h1").childNodes[1].textContent}`;
+		} else if (pathname.includes("/lyrics")) {
+			presenceData.largeImageKey = document.querySelector<HTMLMetaElement>(
+				'[property="og:image"]'
+			).content;
+			presenceData.smallImageKey = "https://i.imgur.com/Yb4CWnm.png";
+			presenceData.details = "Reading lyrics";
+			presenceData.state = `${
+				document.querySelector("#site h2 > span > a")?.textContent ||
+				document.querySelector("#site h2 > span > span > a").textContent
+			} - ${document.querySelector("#site h1").childNodes[1].textContent}`;
+		} else if (pathname.includes("/artist")) {
+			const avatar = document.querySelector<HTMLMetaElement>(
+				'[property="og:image"]'
+			).content;
+			if (!avatar.includes("avatar-placeholder.png")) {
+				presenceData.largeImageKey = avatar;
+				presenceData.smallImageKey = "https://i.imgur.com/Yb4CWnm.png";
+			}
+			presenceData.details = "Browsing artist";
+			presenceData.state = document.querySelector(
+				"#site > #artist-page > #content h1"
+			).textContent;
+		}
+	} else if (hostname === "curators.musixmatch.com") {
+		if (pathname === "/" || pathname === "/missions")
+			presenceData.details = "Browsing curator's missions";
+		else if (pathname.includes("/tasks/")) {
+			presenceData.details = `Browsing ${
+				document.querySelector(
+					"#__next > div > div > div > div > div > div > div:nth-child(2) > div.css-1dbjc4n.r-1p0dtai.r-1d2f490.r-12vffkv.r-u8s1d.r-zchlnj.r-ipm5af > div.css-1dbjc4n.r-13awgt0.r-12vffkv > div > div > div > div > div > div.css-1dbjc4n.r-1mdbw0j.r-1guathk > div > div.css-1dbjc4n.r-obd0qt.r-18u37iz.r-1wtj0ep.r-1wzrnnt > div.css-901oao.r-jwli3a.r-1grxjyw.r-uho16t.r-146t82q.r-tskmnb.r-fdjqy7"
```

Comment:
Can you shorten this?

Reply:
i can replace it and get this information from title (something like this) ![image](https://user-images.githubusercontent.com/63014799/185162765-c1088049-810b-4d54-bf57-803ab1309eb5.png)
because idk how to make better selectors here 
![image](https://user-images.githubusercontent.com/63014799/185162580-f8aa8856-037f-4d68-b996-073c7c59dd16.png)

Reply:
Can you link me this page and I'll see what I can do?

Reply:
https://curators.musixmatch.com/missions but probably you can't use it without musixmatch account with curator status

Reply:
Ah no worries then